testsuite/tests/singlecluster/defaults/test_rules_exclusivity.py

```diff
@@ -0,0 +1,44 @@
+"""Test mutual exclusivity of defaults block and implicit defaults"""
+
+import pytest
+from openshift_client import OpenShiftPythonException
+
+from testsuite.kuadrant.policy.rate_limit import Limit
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.limitador]
+
+
+@pytest.fixture(scope="module")
+def authorization(authorization, oidc_provider):
+    """Create AuthPolicy with basic oidc rules inside and outside defaults block"""
+    authorization.defaults.identity.add_oidc("inside-defaults", oidc_provider.well_known["issuer"])
+    authorization.identity.add_oidc("outside-defaults", oidc_provider.well_known["issuer"])
+    return authorization
+
+
+@pytest.fixture(scope="module")
+def rate_limit(rate_limit):
+    """Add basic rate limiting rules inside and outside defaults block"""
+    rate_limit.defaults.add_limit("inside-defaults", [Limit(2, 5)])
+    rate_limit.add_limit("outside-defaults", [Limit(2, 5)])
+    return rate_limit
+
+
+@pytest.fixture(scope="module")
+def commit():
+    """We need to try to commit objects during the actual test"""
+    return None
+
+
+@pytest.mark.parametrize(
+    "component_fixture",
+    [
+        pytest.param("authorization", id="AuthPolicy"),
+        pytest.param("rate_limit", id="RateLimitPolicy"),
+    ],
+)
+def test_rules_exclusivity(request, component_fixture):
```

Comment:
This is too confusing. I would rather haev two test cases with creation of the policies inside it as the act of creation is part of the testcase and as such should be done in test function and not in fixtures

Reply:
Difference between this test and most other tests is that most other tests expect policy to exist and only test behaviour afterwards, hence creation is not important and is not part of the testfunction. This test tests the creation itself and as such is much more similar to `test_gw_doesnt_exists` and similar reconciliation tests. In these test the creation of policy itself is the tested behaviour and should be present in the test function and not in the fixture, making it hard to understand what is going on for someone who is not familiar with pytest internals.

TLDR: For this test I think it would be better to 1) simplify it 2) not use fixtures for policies and instead create them directly in the test function

Reply:
Makes sense. Thanks for the explanation